### ISSUE TITLE
Metadata endpoint response does not contain asset id

### ISSUE THREAD

**Issue by manolodewiner** [bigchaindb/bigchaindb#1896]
Metadata call returns all metadata that match a given text search along with the id of the metadata which is the id of the transaction where it was defined.
For some use cases (for instance when the metadata represents the state of an asset), it would be nice if we also include the asset id related to a transaction. 

So instead of:
```json
[
    {
        "metadata": {"state": "sold"},
        "id": "51ce82a14ca274d43e4992bbce41f6fdeb755f846e48e710a3bbb3b0cf8e4204"
    },
   {
        "metadata": {"state": "created"},
        "id": "b4e9005fa494d20e503d916fa87b74fe61c079afccd6e084260674159795ee31"
    }
]
```
The response could be:
```json
[
    {
        "metadata": {"state": "sold"},
        "id": "51ce82a14ca274d43e4992bbce41f6fdeb755f846e48e710a3bbb3b0cf8e4204",
        "asset_id": "8b20dbe164badd5ca0611b0e233aef9acce609fbca20f787fc7d926f300d0102",

    },
    {
        "metadata": {"state": "created"},
        "id": "b4e9005fa494d20e503d916fa87b74fe61c079afccd6e084260674159795ee31",
        "asset_id": "d07285a60352838ff263a46ba8cca64e18d36888aee0ba76d5d601137b492fc6"
    }
]
```

**Comment by ttmc** [comment]
My comment was that one can get the `"asset_id"` from the transaction, which you _can_ get because you have the transaction ID.

Maybe you want to enhance the metadata endpoint in the HTTP API, so that you can specify the asset_id in addition to the text search? In other words, "Find me all the metadata with this text string and associated with this asset." To do that, I think we'd also want a new index.

**Comment by manolodewiner** [comment]
Yes @ttmc you are right. To search by metadata text and asset id would be useful.
For exampleiIn the RBAC model the _can_link_ is included in the metadata and in that way you could search for the public keys included in the metadata for a specific asset id.

**Comment by kansi** [comment]
@ttmc This issue will introduce a new API i.e. `/api/v1/metadata?search=_&asset_id=_`. This means that when extracting and storing metadata we would also need to store the `asset_id` with it and create an index on it. This will be a breaking change and will require a new release.
